x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/PlannerUtils.java

```diff
@@ -139,16 +121,41 @@ public static String[] planOriginalIndices(PhysicalPlan plan) {
             return Strings.EMPTY_ARRAY;
         }
         var indices = new LinkedHashSet<String>();
-        plan.forEachUp(
-            FragmentExec.class,
-            f -> f.fragment().forEachUp(EsRelation.class, r -> addOriginalIndexIfNotLookup(indices, r.index()))
-        );
+        forEachFromRelation(plan, relation -> indices.addAll(asList(Strings.commaDelimitedListToStringArray(relation.index().name()))));
         return indices.toArray(String[]::new);
     }
 
-    private static void addOriginalIndexIfNotLookup(Set<String> indices, EsIndex index) {
-        if (index.indexNameWithModes().get(index.name()) != IndexMode.LOOKUP) {
-            indices.addAll(asList(Strings.commaDelimitedListToStringArray(index.name())));
+    /**
+     * Iterates over the plan and applies the action to each {@link EsRelation} node.
+     * <p>
+     *     This method ignores the right side of joins.
+     * </p>
+     */
+    private static void forEachFromRelation(PhysicalPlan plan, Consumer<EsRelation> action) {
```

Comment:
Extracted the `planConcreteIndices()` code here, and fixed it by adding the `node instanceof Join` check.
Looks like the Join was outside the fragment in the tests I used for the original PR, but it was inside for this case.